spyder/api/plugin_registration/registry.py

```diff
@@ -170,6 +202,17 @@ def _instantiate_spyder5_plugin(
         else:
             self.internal_plugins |= {plugin_name}
 
+        if external:
+            if not running_under_pytest():
+                # These attributes come from spyder.app.find_plugins
+                module = PluginClass._spyder_module_name
+                package_name = PluginClass._spyder_package_name
+                version = PluginClass._spyder_version
+                description = instance.get_description()
+                dependencies.add(module, package_name, description,
+                                    version, None,
+                                    kind=dependencies.PLUGIN)
```

Comment:
This exact same code is part of the main window:

https://github.com/spyder-ide/spyder/blob/c23667f46f967338ce447084deab8254799b5319/spyder/app/mainwindow.py#L902-L909

Should it be removed from there to only leave it here? That would avoid registering external plugins twice.

----

Also, I think it's not necessary to have now `if not running_under_pytest():` because we're installing `spyder_boilerplate` to test it on CIs.